process_email.py

```diff
@@ -622,35 +620,50 @@ def _handle_attachment(self, part, file_name, file_path):
                 del attach_meta_info['content']
                 curr_attach['should_ignore'] = True
 
-        part_payload = part.get_payload(decode=True)
-        try:
-            if not part_payload:
-                with open(file_path, 'wb'):
-                    self._base_connector.debug_print("Payload not found, hence adding empty file")
-            else:
+        if isinstance(part.get_payload(), list):
+            try:
+                part_payload = part.get_payload()[0]
```

Comment:
I think this is the only line that actually raises `IndexError` so why is the whole block inside `try`?

Reply:
Moved writing to the file outside of the try block